## Analyse der MedReg Daten über Psychiater

In diesem Notebook sollen die bereinigten und überprüften Daten aus den beiden Notenooks [MedReg_Analyse_01_Bereinigen](/Users/master/Desktop/CAS_DJ/MedReg/MedReg Scraping 2020/MedReg_Analyse_01_Bereinigen.ipynb) und [MedReg_Analyse_02_Datencheck](/Users/master/Desktop/CAS_DJ/MedReg/MedReg Scraping 2020/MedReg_Analyse_02_Datencheck.ipynb) einer weiteren Analyse unterzogen werden. 

Mit den bereinigten Daten sollen hier folgende Auswertungen wiederholt oder neu gemacht werden:

1. Stand Facharztabschlüsse Psychiatrie und Physotheraphie nach Nationalität per Ende 2019 (auf Basis von Land Medizinsudium und der direkten Anerkennungen ausländischer Facharzttiteln)
2. Aussagen über die Nationalität bzw. Studienort der anerkannten Weiterbildungstitel
3. Entwicklung Anteil der Anerkennung (sprich: Neuzulassung) per Jahresende, über die Jahre, bis Ende 2019


Für diese Analysen werde ich die Daten **aller** registrierten Arzttitel verwenden und nicht bloss jene mit erteilter Bewilligung, da diese nur für eine bestimmte Art von (selbstständigen) Tätigkeiten nötig ist. So schrieb das BAG auf eine Anfrage im letzten Herbst: *«Die Statistiken des BAG zu den aktiven Bewilligungen nach MedBG 2018 resultieren aus den von den Kantonen erteilten Berufsausübungsbewilligungen. Eine solche Bewilligung ist aufgrund des Medizinalberufegesetzes (MedBG) für die privatwirtschaftliche Berufsausübung in eigener fachlicher Verantwortung notwendig. Es sind somit nicht alle Tätigkeiten bewilligungspflichtig. Entsprechend kann das BAG keine Aussagen dazu machen, wie viele Ärzt/innen in der Schweiz ihren Beruf tatsächlich ausüben.»*


In weiteren Analysen sollen später auch Fragen untersucht werden, wie:

Weitere Analysen:
- regionale Aussagen über die Tätigkeitsorte (Adresse der Bewilligung) wie zum Beispiel:
- Vergleich zur Kantonsbevölkerung? (aussagekräftig?)
- Städtische Gebiete, ländliche Gebiete? (aufwändig)
- Grossstädte vs. landliche Zentren?

In [ ]:
import pandas as pd
import re
import numpy as np
import datetime

In [ ]:
df = pd.read_csv(r'medreg_2020_psych_b2019_20200128.csv')

In [ ]:
len(df)

In [ ]:
df_erteilt = df[df['BAB Status'] == 'Erteilt']

In [ ]:
len(df_erteilt)

Wir haben also insgesamt 5413 registrierte Fachärzte per Ende 2019 im MedReg. Diese Zahl erscheint plausibel, waren es doch am 7.1.2020 5420 Einträge [Screenshot](/Users/master/Desktop/CAS_DJ/MedReg/Screenshots MedReg/Screenshot MedReg Stand 20200107 at 002505.jpg)


## 1. Analyse: Herkunft der Psychiaterinnen und Psychiater nach dem Land ihres Medizinstudiums


In [6]:
df.dtypes

Unnamed: 0                 int64
Unnamed: 0.1               int64
Doc_ID                     int64
Name                      object
Geschlecht                object
Adresse                   object
Ort                       object
Nationalität              object
Sprache                   object
Arzt_Jahr                  int64
Arzt_Land                 object
Arzt_Anerkennung_Jahr    float64
PP_Jahr                    int64
PP_Land                   object
PP_Anerkennung             int64
BAB                       object
BAB Status                object
Datei                     object
PP_CH                      int64
dtype: object

In [7]:
df_arzt_land = df['Arzt_Land'].value_counts()

In [8]:
df_arzt_land = df_arzt_land.reset_index()

In [11]:
df_arzt_land.head(10)

index  Arzt_Land
0            Schweiz       2833
1        Deutschland       1354
2         Frankreich        316
3            Italien        213
4         Österreich        179
5            Spanien         73
6           Rumänien         61
7       Griechenland         57
8  vorm. Jugoslawien         42
9            Belgien         41

In [24]:
df_arzt_land = df_arzt_land.rename(columns = {'index':'Land Medizinstudium', 'Arzt_Land':'Anzahl'})

In [25]:
df_arzt_land['Anteil'] = df_arzt_land['Anzahl'] / 5413 * 100

In [26]:
df_arzt_land.to_csv(r'MedReg_Analyse_output/Psych_Land_Medizinstudium_bis2019.csv',index = 'Land Medizinstudium', header=True)

FileNotFoundError: [Errno 2] No such file or directory: 'MedReg_Analyse_output/Psych_Land_Medizinstudium_bis2019.csv'

In [ ]:
df_arzt_land.head(1

### Resultat 1: 52.2% aller in der Schweiz registrierten Psychiaterinnen und Psychiater haben ihr Grundstudium in der Schweiz absolviert. 25% haben einen deutschen Abschluss, 5.8% einen französischen. (Stand: Ende 2019).
Wir möchten diese Zahl noch in den Kontext mit anderen Fachrichtungen stellen. Dazu müssen wir aber den gesamten Datensatz aus dem Scraping nochmals aufbereiten. Dies soll zu einem späteren Zeitpunkt nachgeholt werden, falls es die Zeit erlaubt. Als Vergleich schauen wir uns noch kurz die Nationalität der registrierten Ärzte an. Allerdings ist dieser Wert meiner Meinung nach weniger aussagekräftig, da er durch Einbürgerungen verwässert wird. 

In [18]:
df_nation = df['Nationalität'].value_counts()

In [19]:
df_nation = df_nation.reset_index()

In [20]:
df_nation = df_nation.rename(columns = {'index':'Nationlität', 'Nationalität':'Anzahl'})

In [21]:
df_nation['Anteil'] = df_nation['Anzahl'] / 5413 * 100

In [23]:
df_nation.head(10)

Nationlität  Anzahl     Anteil
0        Schweiz (CH)    2938  54.276741
1    Deutschland (DE)    1358  25.087752
2     Frankreich (FR)     355   6.558286
3        Italien (IT)     188   3.473120
4     Österreich (AT)     162   2.992795
5   Griechenland (GR)      71   1.311657
6        Spanien (ES)      66   1.219287
7       Rumänien (RO)      44   0.812858
8        Belgien (BE)      35   0.646592
9      Bulgarien (BG)      15   0.277111

In [ ]:
df_nation.to_csv(r'MedReg_Analyse_output/Psych_Nationalitaet_bis2019.csv',index = 'Nationalität', header=True)

### Restultat 2: nach Nationalität ergibt sich ein ähnliches Bild. Danach sind 54.3% Schweizer, 25.1% Deutsche und 6.6% Franzosen.

Zum Vergleich wollen wir noch die Daten aller Ärzte mit erteilter Bewilligung auswerten. Bei diesen ist vermutlich der Anteil der niedergelassenen, selbstständig tätigen Ärzte höher. 

In [ ]:
df_erteilt_arzt_land = df_erteilt['Arzt_Land'].value_counts()

In [ ]:
df_erteilt_arzt_land = df_erteilt_arzt_land.reset_index()

In [ ]:
df_erteilt_arzt_land.head(5)

In [ ]:
df_erteilt_arzt_land = df_erteilt_arzt_land.rename(columns = {'index':'Land Medizinstudium', 'Arzt_Land':'Anzahl'})

In [ ]:
df_erteilt_arzt_land['Anteil'] = df_erteilt_arzt_land['Anzahl'] / 4415 * 100

In [ ]:
df_erteilt_arzt_land.to_csv(r'MedReg_Analyse_output/Psych_BAB_Land_Medizinstudium_bis2019.csv',index = 'Land Medizinstudium', header=True)

In [ ]:
df_erteilt_arzt_land.head(5)

### Restultat 3: mit erteilter Bewilligung liegt der Schweizer Anteil höher: 57.6% CH-Medizinstudium, 22.4% Deutschland. 

Jetzt schauen wir genauer hin. Wie war das mit den neuzugelassenen Psychiatern im letzten Jahr?

In [ ]:
df.head(2)

In [ ]:
df2019 = df[df['PP_CH'] == 2019]

In [ ]:
total = len(df2019)

In [ ]:
df2019 = df2019['Arzt_Land'].value_counts().reset_index().rename(columns = {'index':'Land Medizinstudium', 'Arzt_Land':'Anzahl'})

In [ ]:
df2019.head(2)

In [ ]:
df2019['Anteil'] = df2019['Anzahl'] / int(total) * 100

In [ ]:
df2019.head(5)

In [ ]:
df2019.to_csv(r'MedReg_Analyse_output/Psych_BAB_Land_Medizinstudium_2019.csv',index = 'Land Medizinstudium', header=True)

### Restultat 4: 2019 waren von den 183 neu registrierten Fachärzten Psychiatrie und Psychotherapie nur 35 darunter, die in der Schweiz Medizin studiert hatten. Das sind bloss 20%! 30% hatten ein deutsches Medizinstudium, 15% ein französisches. Ein Drittel verteilt sich auf die weiteren Länder. 

## 2. Analyse: Entwicklung Anteil der Anerkennung (sprich: Neuzulassung) per Jahresende, über die Jahre
Wir möchten uns jetzt anschauen, wie sich dieser Anteil über die letzten Jahre entwickelt hat. 

Es muss uns dabei bewusst sein, dass wir dabei die Daten eines dynamischen Registers verwenden und es ein Blick zurück vom Stand Ende 2019 ist. Das heisst: je weiter wir zurückblicken, desto mehr sind die Zahlen durch Mutationen verwässert.

Zuerst ein Überblick über die Anzahl Zulassungen über die letzten Jahre. Wieviele Diplome gingen jeweils an Medizinabsolventen mit Schweizer Arztdiplom?


In [ ]:
df[df['Arzt_Land'] == "Schweiz"].groupby('PP_CH')['PP_CH'].count()

In [ ]:
%matplotlib inline

In [ ]:
df[df['Arzt_Land'] == "Schweiz"].groupby('PP_CH')['PP_CH'].count().plot(figsize=(15,8))

Und jetzt nehmen wir noch die Daten aller ohne schweizerisches Medizinstudium hinzu:

In [ ]:
chart1 = df[df['Arzt_Land'] == "Schweiz"].groupby('PP_CH')['PP_CH'].count().plot(figsize=(15,8))
df[df['Arzt_Land'] != "Schweiz"].groupby('PP_CH')['PP_CH'].count().plot(ax=chart1)

Nächster Schritt: wir wollen jetzt die Daten aller Schweizer und ausländischen Medizinstudium-Absolventen in ein neues Dataframe überführen, zur Auswertung. 

In [ ]:
df_all = pd.DataFrame()

In [ ]:
df_all['CH'] = df[df['Arzt_Land'] == "Schweiz"].groupby('PP_CH')['PP_CH'].count()
df_all['Ausland'] = df[df['Arzt_Land'] != "Schweiz"].groupby('PP_CH')['PP_CH'].count()
df_all['Total'] = df_total = df.groupby('PP_CH')['PP_CH'].count()

In [ ]:
df_all = df_all.reset_index()

In [ ]:
df_all.head(3)

In [ ]:
df_all.rename(columns = {'PP_CH' : 'Jahr'})

In [ ]:
df_all['CH %'] = df_all['CH'] / df_all['Total'] * 100

In [ ]:
df_all['Ausl %'] = df_all['Ausland'] / df_all['Total'] * 100

In [ ]:
df_all.tail(20)

In [ ]:
df_all = df_all.rename(columns = {'PP_CH' : 'Jahr'})

In [ ]:
df_plot1 = df_all[['Jahr', 'CH %', 'Ausl %']]

In [ ]:
df_plot1 = df_plot1.tail(15)

In [ ]:
df_plot1.plot(x = 'Jahr', y = ['CH %', 'Ausl %'], kind = 'area', grid = True, fontsize = 18, figsize=(20,10))

In [ ]:
df_all.to_csv(r'MedReg_Analyse_output/Psych_Land_Medizinstudium_2000b2019.csv',index = 'Jahr', header=True)

### Resultat 5: Der Gesamtanteil an neu zugelassenen Fachärzten 'Psychiatrie und Psychotherapie' ist in den letzten Jahren gesunken,  im Vergleich zur Vordekade hat er sich seit 2010 gut halbiert. 
Wir möchten jetzt noch die Daten der direkt anerkannten Diplome separat der Tabelle hinzufügen (sie sind aber bereits in den Daten der Spalte 'Ausland' berücksichtigt). Besonders seit 2012 ist hier ein grosser Anstieg zu verzeichnen, zeigte bereits eine frühere Auswertung von BAG-Daten.

In [ ]:
df_anerkannt = df.groupby('PP_Anerkennung')['PP_Anerkennung'].count()

In [ ]:
df_anerkannt = df_anerkannt.to_frame(name='Anerkannt')

In [ ]:
df_anerkannt.reset_index(inplace = True)

In [ ]:
df_anerkannt = df_anerkannt.rename(columns = {'PP_Anerkennung' : 'Jahr'})

In [ ]:
df_anerkannt = df_anerkannt.drop([0], axis=0)

In [ ]:
df_all2 = pd.merge(df_all, df_anerkannt, how='outer', on='Jahr')

In [ ]:
df_all2.tail(3)

In [ ]:
df_2002 = df_all2.tail(18)

In [ ]:
df_2002

In [ ]:
import matplotlib.pyplot as plt

In [ ]:
df_2002.plot(x="Jahr", y=["CH", "Ausland", "Anerkannt"], grid = True, linewidth=3, fontsize = 18, figsize=(20,10))

## Check: Quervergleich
Wir wollen diese Analyse noch mit den vom BAG zur Verfügung gestellten Daten aus dem Jahr 2018 vergleichen. Allerdings sind dort nur alle Einträge mit erteilter Berufsausübungsbewilligung registriert. 

In [ ]:
df_bag2018 = pd.read_excel(r'/Users/master/Desktop/CAS_DJ/MedReg/Daten_BAG/BAG Aktive Ärzte Auszug MedReg/rohdaten-aktive-bab-med-2018_mod.xlsx')

In [ ]:
df_bag2018.head(2)

In [ ]:
df_bag2018_psych = df_bag2018[df_bag2018['Facharzttitel'] == 'Psychiatrie und Psychotherapie']

In [ ]:
len(df_bag2018_psych)

In [ ]:
df_bag2018_psych['JahrAnerkennungWBT'].value_counts()

In [ ]:
df_bag2018_psych['JahrGleichwertigkeitWBT'].value_counts()

Check über alle Ärztegruppen:

In [ ]:
df_bag2018['JahrGleichwertigkeitWBT'].value_counts()

Dieses Feld erscheint praktisch obsolet, über alle Fachrichtungen wurde bloss ein Eintrag im Jahr 2018 darin festgehalten. 

## Check 1: Vergleich mit Zahlen 2018 des BAG
Wir müssen die Daten nochmals überprüfen. Zumindest bei erteilter Berufsbewilligung kommen wir auf tiefere Zahlen, was die Anerkennung von Weiterbildungstiteln betrifft. 
Wir vergleichen dazu nochmals die Daten aus dem Scraping, aber nehmen nur jene mit erteilter Bewilligung. 



In [ ]:
df.head(3)

In [ ]:
df_erteilt = df[df['BAB Status'] == 'Erteilt']

In [ ]:
len(df_erteilt)

In [ ]:
df_erteilt['PP_Anerkennung'].value_counts()

In [ ]:
df['PP_Anerkennung'].value_counts()

Also: wir kommen bei allen Psychiatern mit erteilter kantonaler Bewilligung für das Jahr 2018 auf 47 anerkannte Diplome, 2017: 75, 2016: 83, 2015: 86. 

Die BAG-Datei 2018 liefert uns für 2015 ebenfalls 86 Anerkennungen, für 2016: 72 (-9), für 2017: 54 (-21), für 2018: bloss 24 (-23). 

Die Zahlen liegen näher, je weiter die Anerkennung zurückliegt. **Dies erscheint schlüssig: werden sich doch Ärztinnen und Ärzte erst nach der Anerkennung ihrer Titel bei einem Kanton um eine Bewilligung bemühen, wenn sie sich zum Beispiel mit einer eigenen Praxis niederlassen wollen.** Deshalb kommt es vermutlich zu diesem verzögerten Effekt. 

## Check 2: Anerkannte Weiterbildungstitel
Um der Grundlage unserer Datenanalyse sicher zu sein, prüfen wir noch eine weitere Datenquelle des BAG, die wir bereits in einem frühren Schritt ausgewertet haben. Es handelt sich dabei um die Datei *uebersicht-anerkannte-wbt-humanmedizin-2018* über die Anerkennung ausländischer Facharzttitel [Link](https://www.bag.admin.ch/dam/bag/de/dokumente/berufe-gesundheitswesen/medizinalberufe/statistiken/med/uebersicht-anerkannte-wbt-humanmedizin-2018.xlsx.download.xlsx/uebersicht-anerkannte-wbt-humanmedizin-2018.xlsx). 

Diese [Auswertung](/DatenBAG_BAG/Anerkennung_Psychiatrie_eigene_Auswertung.csv) ergab bis 2018 folgende Zahlenreihe:
61	74	48	59	53	65	64	53	62	66	107	118	107	139	135	123	93, das heisst:

2018: 93 Anerkennungen (Scrapingdaten: 94, +1)

2017: 123 (130, +7)

2016: 135 (133, -2)

2015: 139 (139, 0)

2014: 107 (108, +1)

Fazit: auch hier sind es sehr wahrscheinlich Mutationen, die zu diesen Unterschieden führt. Die Abweichungen sind eher gering - und wirken sich nicht nur wenig auf das Gesamtbild aus. Die Zahlen aus der Auswertung zeichnen deshalb ein Bild, mit der Einschränkung, dass wir nicht wissen, ob und in welchem Grad die im MedReg registrierten Ärztinnen und Ärzte tätig sind. Das gilt aber für alle gleichermassen. 


